src/Microsoft.AspNetCore.Mvc.Core/Internal/VirtualFileResultExecutor.cs

```diff
@@ -59,14 +69,31 @@ public Task ExecuteAsync(ActionContext context, VirtualFileResult result)
                         count: null,
                         cancellation: default(CancellationToken));
                 }
-                else
+                else if (range == null || !fileStream.CanSeek)
```

Comment:
It may be moot, I don't think you can do your proper range verification without knowing the length.